### PR TITLE
Call into $bulkdata-status passing in the encoded job number

### PR DESCRIPTION
**Pull request by prb112**
The job number is not decoded before used. #2128

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java [hunk 1]
```diff
@@ -26,18 +29,33 @@
...
+    public static JobIdEncodingTransformer getInstance() {
+        if (transformer == null) {
+            transformer = new JobIdEncodingTransformer();
+        }
+        // final SecretKeySpec BATCHJOBID_ENCRYPTION_KEY = getJobIdEncryptionKey();
+        return transformer;
+    }
+
     /*
      * gets the server-wide specific encryption key.
```
Comment (lmsurpre) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:55]: is this comment stale now?

File: operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java [hunk 2]
```diff
@@ -65,15 +83,17 @@ private static SecretKeySpec getJobIdEncryptionKey() {
      * @param jobId
      * @return
      */
-    public String endcodeJobId(String jobId) {
```
Comment (lmsurpre) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:68]: good typo fix

File: operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/StatusOperation.java [hunk 3]
```diff
@@ -55,7 +55,6 @@ protected Parameters doInvoke(FHIROperationContext operationContext, Class<? ext
         if (logicalId == null && versionId == null && resourceType == null) {
             String method = (String) operationContext.getProperty(FHIROperationContext.PROPNAME_METHOD_TYPE);
             if ("DELETE".equalsIgnoreCase(method)) {
-                // Assume GET or POST
```
Comment (lmsurpre) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/StatusOperation.java:58]: good comment fix

File: operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java (hunk 1 above)
Comment (prb112) [reply operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:55]: Yes, I'll c lean up

Comment (prb112) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:55]:
```suggestion
     * gets the tenant. specific encryption key.
```

File: operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java [hunk 4]
```diff
@@ -26,18 +29,33 @@
...
+    /**
+     * get the instance
+     *
+     * @return
+     */
+    public static JobIdEncodingTransformer getInstance() {
+        if (transformer == null) {
+            transformer = new JobIdEncodingTransformer();
+        }
+        // final SecretKeySpec BATCHJOBID_ENCRYPTION_KEY = getJobIdEncryptionKey();
```
Comment (lmsurpre) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:50]: missed this on my first pass through...would prefer commented out code to be deleted (if not, then a comment explaining what it is and why its commented out)

Comment (prb112) [reply operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:50]: good catch, not suppose to. be there now. I'll remove

Comment (prb112) [inline operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java:50]:
```suggestion
```

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**